protostar/starknet_gateway/block_explorer/block_explorer_factory.py

```diff
@@ -0,0 +1,29 @@
+from typing import Literal, Optional
+
+from ..network_config import PredefinedNetwork
+from .block_explorer import BlockExplorer
+from .stark_scan_block_explorer import StarkScanBlockExplorer
+from .viewblock_block_explorer import ViewblockBlockExplorer
+from .voyager_block_explorer import VoyagerBlockExplorer
+from .fake_block_explorer import FakeBlockExplorer
+
+SupportedBlockExplorerName = Literal["starkscan", "viewblock", "voyager"]
+SUPPORTED_BLOCK_EXPLORER_NAMES: list[SupportedBlockExplorerName] = [
+    "starkscan",
+    "viewblock",
+    "voyager",
+]
+
+
+def create_block_explorer(
+    block_explorer_name: SupportedBlockExplorerName,
+    network: Optional[PredefinedNetwork],
```

Comment:
I think `network` could be just an argument to all `create_*` functions and thus you could make `BlockExplorer` instances parameterless.

Reply:
I'd have to then pass network to formatters.

Reply:
OK, makes sense